tools/common/StaticRegistrar.cs

```diff
@@ -3296,9 +3296,9 @@ void Specialize (AutoIndentStringBuilder sb, out string initialization_method, s
 			sb.WriteLine (map.ToString ());
 			sb.WriteLine (map_init.ToString ());
 
-			if (!string.IsNullOrEmpty (type_map_path)) {
-				var doc = CSToObjCMap.ToXDocument (map_dict);
-				doc.Save (type_map_path);
+			if (rewriteClassHandles) {
+				var rewriter = new Rewriter (map_dict, GetAssemblies ());
+				rewriter.Process ();
```

Comment:
IIRC the class redirect is .NET only, and not for legacy Xamarin:

```suggestion
#if NET
			if (rewriteClassHandles) {
				var rewriter = new Rewriter (map_dict, GetAssemblies ());
				rewriter.Process ();
			}
#endif
```